pkg/etcd/etcd.go

```diff
@@ -1067,15 +1077,33 @@ func (e *ETCD) manageLearners(ctx context.Context) {
 			logrus.Warnf("Failed to list nodes with etcd role: %v", err)
 		}
 
+		// a map to track if a node is a member of the etcd cluster or not
+		nodeIsMember := make(map[string]bool)
+		nodesMap := make(map[string]*v1.Node)
+		for _, node := range nodes {
+			nodeIsMember[node.Name] = false
+			nodesMap[node.Name] = node
+		}
+
 		for _, member := range members.Members {
 			if member.IsLearner {
 				if err := e.trackLearnerProgress(ctx, progress, member); err != nil {
 					logrus.Errorf("Failed to track learner progress towards promotion: %v", err)
 				}
 				for _, node := range nodes {
 					if strings.HasPrefix(member.Name, node.Name+"-") {
-						_, _, err := e.setEtcdStatusCondition(node, client.CoreV1(), member.Name, false)
-						if err != nil {
+						nodeIsMember[node.Name] = true
+
+						// verify if the member is healthy and set the etcd status condition
+						if _, err := e.getETCDStatus(ctx, member.ClientURLs[0]); err != nil {
+							logrus.Errorf("Could not get status for member %s: %v", member.Name, err)
+							if err := e.setEtcdStatusCondition(node, client, member.Name, StatusUnhealthy); err != nil {
+								logrus.Errorf("Unable to set etcd status condition for unhealthy node %s: %v", member.Name, err)
+							}
+							continue
+						}
+
```

Comment:
Don't do this here - we should only check for health once it's been promoted to voting member. If it is unhealthy and fails to be promoted, `trackLearnerProgress` will remove it from the cluster for us.
```suggestion
```